src/pages/tasks/common/hooks.tsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Link } from '@invoiceninja/forms';
+import { date } from 'common/helpers';
+import { route } from 'common/helpers/route';
+import { useFormatMoney } from 'common/hooks/money/useFormatMoney';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { useCurrentCompanyDateFormats } from 'common/hooks/useCurrentCompanyDateFormats';
+import { useCurrentUser } from 'common/hooks/useCurrentUser';
+import { Task } from 'common/interfaces/task';
+import { customField } from 'components/CustomField';
+import { DataTableColumnsExtended } from 'pages/invoices/common/hooks/useInvoiceColumns';
+import { useTranslation } from 'react-i18next';
+import { TaskStatus } from './components/TaskStatus';
+import { calculateEntityState } from './helpers/calculate-entity-state';
+import { calculateTime } from './helpers/calculate-time';
+
+export const taskColumns = [
+  'status',
+  'number',
+  'client',
+  //   'project',  @Todo: Need to fetch relationship
+  'description',
+  'duration',
+  'entity_state',
+  'archived_at',
+  //   'assigned_to', @Todo: Need to fetch relationship
+  //   'calculated_rate', @Todo: Diff vs `rate`?
+  'created_at',
+  //   'created_by', @Todo: Need to fetch relationship
+  'custom1',
+  'custom2',
+  'custom3',
+  'custom4',
+  //   'date', @Todo: What column does this map to?
+  //   'documents', @Todo: Does this map to `invoice_documents`?
+  //   'invoice', @Todo: Need to fetch the relationship
+  'is_deleted',
+  'is_invoiced',
+  //   'is_running', @Todo: Need manual calc
```

Comment:
@turbo124 Can you pass me block from the backend to check if task is running?

Reply:
we calculate task is running by checking the last array, if the second value is 0 the task is running, ie.

```
[[1,2],[3,4]] - task not running
```
```
[[1,2],[3,0]] - task not running
```